app/services/github/contributions/wrappers/pull_request.rb

```diff
@@ -16,12 +19,78 @@ def repository_name
           @repository_name ||= pull_request.html_url.split("/")[-4..-3].join("/")
         end
 
+        def repository_owner
+          @repository_owner ||= pull_request.html_url.split("/")[-4]
+        end
+
+        def pull_request_number
+          @pull_request_number ||= pull_request.html_url.split("/")[-1]
+        end
+
+        def pull_request_commits
+          @pull_request_commits ||= client.pull_requests.commits(
+            user: repository_owner, repo: repository_name.split("/")[-1], number: pull_request_number)
+        end
+
+        def get_co_authors_data_by_commit_messages
+          co_authors_data = pull_request_commits.map do |commit|
+            commit_message = commit.dig("commit", "message")
+
+            data = nil
+
+            unless !commit_message.match?(CO_AUTHOR_REGEX)
+              data = commit_message.scan(CO_AUTHOR_REGEX).flatten
+              data.first.strip!
+            end
+
+            data
+          end.compact
```

Comment:
Perhaps you could replace `map {...}.compact` in favor of `filter_map`.
```suggestion
          co_authors_data = pull_request_commits.filter_map do |commit|
            commit_message = commit.dig("commit", "message")

            data = nil

            unless !commit_message.match?(CO_AUTHOR_REGEX)
              data = commit_message.scan(CO_AUTHOR_REGEX).flatten
              data.first.strip!
            end

            data
          end
```

https://www.rubydoc.info/stdlib/core/Enumerable:filter_map